### PR TITLE
Issue 3 user can edit an entry

### PR DESCRIPTION
**Pull request by jooliecodes**
https://github.com/julieluangpraxay/code-journal/assets/130415142/296c0429-75c1-4d9e-9c87-faa6b6199cb3

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -18,26 +17,45 @@ $submitForm.addEventListener('submit', function (event) {
...
+      ) {
+        $liElements[i].replaceWith(renderEntry(formData));
+      }
+    }
+
+    $entryTitle.textContent = 'New Entry';
+    viewSwap('entries');
+    data.editing = null;
+  }
 });
```
Comment (shawnkost) [inline js/main.js]: `viewSwap` and `toggleNoEntries` should be getting called here instead

File: js/main.js [hunk 2]
```diff
@@ -18,26 +17,45 @@ $submitForm.addEventListener('submit', function (event) {
...
-  $image.src = './images/placeholder-image-square.jpg';
-  $submitForm.reset();
-  toggleNoEntries();
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(formData);
+    $ul.prepend(renderEntry(formData));
+    $image.src = './images/placeholder-image-square.jpg';
+    $submitForm.reset();
+    toggleNoEntries();
```
Comment (shawnkost) [inline js/main.js]: This will be deleted after the other change

File: js/main.js [hunk 3]
```diff
@@ -18,26 +17,45 @@ $submitForm.addEventListener('submit', function (event) {
...
+  } else {
+    // Assign the entry id value from data.editing to the new object with the updated form values.
+    data.editing.entryId = formData.entryId;
```
Comment (shawnkost) [inline js/main.js]: This should be swapped

File: js/main.js [hunk 4]
```diff
@@ -18,26 +17,45 @@ $submitForm.addEventListener('submit', function (event) {
...
+    //  Replace the original object in the data.entries array for the edited entry with the new object with the edited data.
+    data.entries[data.entries.length - formData.entryId] = formData;
```
Comment (shawnkost) [inline js/main.js]: This won't always work. I suggest using another method of finding the entry you want to update,  like looping over it instead

File: js/main.js [hunk 5]
```diff
@@ -105,3 +133,33 @@ document.querySelector('a').addEventListener('click', function () {
...
+const $entryTitle = document.querySelector('.entry-title');
+
+function pencilClick(event) {
+  // if pencil is clicked and it is the I element
+  if (event.target.tagName === 'I') {
+    const dataEntryId = event.target
+      .closest('li')
+      .getAttribute('data-entry-id');
+    // Use the viewSwap function to show the form if its true
+    viewSwap('entry-form');
```
Comment (shawnkost) [inline js/main.js]: I would move this down to the bottom of the function, so that we only swap after all the `input`s are filled in

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some changes here Julie!

**Comment by jooliecodes** [comment]
https://github.com/julieluangpraxay/code-journal/assets/130415142/420ed7ec-fb37-4b82-a8db-6910f84e7047

File: js/main.js [hunk 6]
```diff
@@ -24,15 +24,16 @@ $submitForm.addEventListener('submit', function (event) {
     data.nextEntryId++;
     data.entries.unshift(formData);
     $ul.prepend(renderEntry(formData));
-    $image.src = './images/placeholder-image-square.jpg';
```
Comment (shawnkost) [inline js/main.js]: This should stay

File: js/main.js [hunk 7]
```diff
@@ -43,11 +44,11 @@ $submitForm.addEventListener('submit', function (event) {
         $liElements[i].replaceWith(renderEntry(formData));
       }
     }
-
     $entryTitle.textContent = 'New Entry';
-    viewSwap('entries');
     data.editing = null;
   }
+  viewSwap('entries');
+  toggleNoEntries();
```
Comment (shawnkost) [inline js/main.js]: You should call `$submitForm.reset()` here instead. This way we always reset the form after creating a new entry or editing an entry

File: js/main.js [hunk 8]
```diff
@@ -24,15 +24,16 @@ $submitForm.addEventListener('submit', function (event) {
...
-    //  Replace the original object in the data.entries array for the edited entry with the new object with the edited data.
-    data.entries[data.entries.length - formData.entryId] = formData;
-    // query for all li elements
+    // new code loop
+    formData.entryId = data.editing.entryId;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === formData.entryId) {
+        data.entries[i] = formData;
+        $image.src = './images/placeholder-image-square.jpg';
+        $submitForm.reset();
```
Comment (shawnkost) [inline js/main.js]: Delete this

**Review by shawnkost — CHANGES_REQUESTED** [review]
Few more changes here Julie!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**